SwiftGit2/CommitIterator.swift

```diff
@@ -0,0 +1,67 @@
+//
+// Created by Arnon Keereena on 4/28/17.
+// Copyright (c) 2017 GitHub, Inc. All rights reserved.
+//
+
+import Result
+import libgit2
+
+public class CommitIterator: IteratorProtocol {
+    public typealias Element = Result<Commit, NSError>
+    var repo: Repository
+    var branch: Branch
+    var revisionWalker: OpaquePointer? = nil
+    var oid: git_oid
```

Comment:
Ok, I see now, after reading the header docs of git_revwalk_push, git_revwalk_next and git_commit_lookup. So the OID needed initially to create the revwalk, then the next git_oid used in git_revwalk_next does not need to be the old, because its git pointed to a new one any way. Then from the oid that is populated. It is used to lookup for the actual commit in git_commit_lookup. After that there's no need for the oid anymore. Thank you for pointing that out. I still understand very less about these things.